client/src/components/Notifications/Broadcasts/BroadcastsOverlay.test.ts

```diff
@@ -9,15 +9,26 @@ import { type BroadcastNotification, useBroadcastsStore } from "@/stores/broadca
 const localVue = getLocalVue(true);
 
 const now = new Date();
-const inTwoMonths = new Date(now.setMonth(now.getMonth() + 2));
+const inTwoMonths = new Date(new Date(now).setMonth(now.getMonth() + 2));
 
-function generateBroadcastNotification(id: string): BroadcastNotification {
+/** API date-time does not have timezone indicator and it's always UTC. */
```

Comment:
That is a good point and we might be missing something in all the API responses when handling datetimes, because as far as I can tell, all responses from all API routes (old and new) are not using the timezone signifier like `"create_time": "2020-06-26T14:22:58.435340"`. I guess we have always serialized the timestamps like this... maybe we should create a different issue to discuss/address this?

Reply:
If you want to update the API to ship tzinfo (Zulu) that'd be great -- I do think times should always be stored and served UTC and localized on the client or receiving end.  Right now, we work around the lack of tzinfo in the utcdate component like this:  https://github.com/galaxyproject/galaxy/blob/dev/client/src/components/UtcDate.vue#L14

Reply:
My main point is that we should likely accept and correctly handle timezone-enabled timestamps coming in API requests. I am happy to create a separate issue if it seems to be wider than the notification framework.

Reply:
Sure, we need to ensure incoming timestamps with timezones are handled correctly. I still need to investigate and fix that part for notifications in a new PR including the UI fixes/enhancements on #17010. This PR only addresses the admin filtering issue in Nate's comment.
What these tests are doing here is generating a fake notification response in the Store, and those apparently are always without Z, that is why I suggest creating a separate issue because it affects every response with a timestamp. I guess that's why we do that trick in the client that Dannon commented.

Reply:
Thanks, sounds good to follow up on the other stuff separately!